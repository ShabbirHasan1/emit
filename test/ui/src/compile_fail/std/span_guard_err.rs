use std::io;

#[emit::span(rt: RT, guard: span, err: (|err| err), "test")]
pub fn exec(fail: bool) -> Result<bool, io::Error> {
    if fail {
        return Err(io::Error::new(io::ErrorKind::Other, "failed"));
    }

    span.complete();

    Ok(true)
}

fn main() {

}
