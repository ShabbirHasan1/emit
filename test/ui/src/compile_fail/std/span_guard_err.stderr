error: the `guard` control parameter is incompatible with `ok_lvl`, `err_lvl`, `panic_lvl`, or `err`
 --> src/compile_fail/std/span_guard_err.rs:1:29
  |
1 | #[emit::span(rt: RT, guard: span, err: (|err| err), "test")]
  |                             ^^^^
